crypto/evp_extra/p_dh.c

```diff
@@ -21,13 +21,18 @@
 
 typedef struct dh_pkey_ctx_st {
   int pad;
+  /* Parameter gen parameters */
+  int prime_len;
+  int generator;
 } DH_PKEY_CTX;
 
 static int pkey_dh_init(EVP_PKEY_CTX *ctx) {
   DH_PKEY_CTX *dctx = OPENSSL_zalloc(sizeof(DH_PKEY_CTX));
   if (dctx == NULL) {
     return 0;
   }
+  dctx->prime_len = 2048;
+  dctx->generator = 2;
```

Comment:
how were these defaults selected? are they valid under FIPS?

Reply:
These are the default values documented by OpenSSL: https://docs.openssl.org/master/man3/EVP_PKEY_CTX_ctrl/#dh-parameters

> EVP_PKEY_CTX_set_dh_paramgen_prime_len() sets the length of the DH prime parameter p for DH parameter generation. If this function is not called then 2048 is used. Only accepts lengths greater than or equal to 256.
...
> EVP_PKEY_CTX_set_dh_paramgen_generator() sets DH generator to gen for DH parameter generation. If not specified 2 is used.

Reply:
sounds good. section 5 of [NIST SP 800-131A](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-131Ar2.pdf) indicates that 2048 is the minimum valid finite field DH key size.